File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        
        HashMap<Character, Integer> hm = new HashMap<>();
        // char val = '\0';

        for (int i = 0; i < s.length(); i++) {
            hm.put(s.charAt(i), hm.getOrDefault(s.charAt(i), 0)+1);
        }

        for (int i = 0; i < s.length(); i++) {
            char val = s.charAt(i);
            if (hm.get(val) == 1)
            return s.indexOf(val);
        }
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.swift
class Solution {
    func firstUniqChar(_ s: String) -> Int {

        // let s = "leetcode"

        // for (index, char) in s.enumerated() {
        //     if s.filter({$0 == char}).count == 1 {
        //         return index
        //     }
        // }
        // return -1

        var dict = [Character : Int]()

        for char in s {
            dict[char, default: 0] += 1
        }

        for (index, char) in s.enumerated() {
            if dict[char] == 1 {
                return index
            }
        }

        return -1
    }
}